src/BlazorWebView/src/Maui/RootComponent.cs

```diff
@@ -14,6 +14,13 @@ namespace Microsoft.AspNetCore.Components.WebView.Maui
 	/// </summary>
 	public class RootComponent
 	{
+		/// <summary>
+		/// Constructs an instance of <see cref="RootComponent"/>.
+		/// </summary>
+		public RootComponent()
+		{
+		}
```

Comment:
Could you clarify why we're adding these empty constructors?

It seems more natural to me to go in the other direction, and make the properties non-nullable and readonly, and only have constructors that involve populating the params.

Is there a use case for uninitialized root component entries?

Reply:
Is it that on some platforms it has to be done this way for compatibility with their editors and setup conventions? If so that's fine for those platforms, but I'm not certain it means we should extend the same looseness to other platforms.

Reply:
For consistency, Maui needs a default constructor in XAML, so its best if we add it to all of them for consistency.

Reply:
Unless it was a single type unified across all three platforms, I'm not convinced that making it consistency error-prone everywhere is desirable.

If it was a single type, then we could argue that people would have reason to write some shared code that operates on this across all hosting platforms. But these are separate types so they don't get that option. I'd rather provide the best usage guidance we can in each case, which means preventing the creation of invalid data where possible.

Reply:
+1 to @SteveSandersonMS . I think providing the least error-prone API here is best. Of course, I think that's what we already had without this PR, which is why I wasn't a fan of doing _anything_, but perhaps we can be a bit more flexible in the API patterns, while still making it hard/impossible to write invalid code.

Reply:
So how about we leave things as they are and close this PR and related issue for that reason.

Reply:
@javiercn - I think if there are reasonable "missing" APIs then I'm fine with adding them. But I don't think the APIs need to be 100% identical on all platforms, especially because that can lead to all kinds of errors. I'm not pushing strongly, though; just my opinion. I'm not strongly against this either.

Reply:
Since we can add more APIs in the future but it's much harder to take them away, I'm in favour of erring on the side of minimizing the API surface. If the existing APIs can be tightened up (to remove any potential invalid cases, like things unwantedly being null) that's great, or if not we can just close this PR and related issue.